src/Build/Utilities/NuGetFrameworkWrapper.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Evaluation
+{
+    /// <summary>
+    /// Wraps the NuGet.Frameworks assembly, which is referenced by reflection.
+    /// </summary>
+    internal class NuGetFrameworkWrapper
+    {
+        /// <summary>
+        /// Resolve the location of the NuGet.Frameworks assembly
+        /// </summary>
+        private static Assembly GetNuGetAssembly()
+        {
+            try
+            {
+                var assemblyDirectory = BuildEnvironmentHelper.Instance.Mode == BuildEnvironmentMode.VisualStudio ?
+                    Path.Combine(BuildEnvironmentHelper.Instance.VisualStudioInstallRootDirectory, "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet") :
+                    BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory;
+                return Assembly.LoadFile(Path.Combine(assemblyDirectory, "NuGet.Frameworks.dll"));
+            }
+            catch
+            {
+                throw new InternalErrorException(AssemblyResources.GetString("NuGetAssemblyNotFound"));
```

Comment:
Worth passing the path where we looked for it to the error? Might speed up an investigation if it's hit.